hazelcast/src/main/java/com/hazelcast/map/impl/query/PartitionScanRunner.java

```diff
@@ -89,7 +93,7 @@ public PartitionScanRunner(MapServiceContext mapServiceContext) {
                 continue;
             }
             //we want to always use CachedQueryEntry as these are short-living objects anyway
-            QueryableEntry queryEntry = new CachedQueryEntry(serializationService, key, value, extractors);
+            QueryableEntry queryEntry = new LazyMapEntry(key, value, serializationService, extractors);
```

Comment:
This is so that we can use the identity projection with the query engine since `CachedQueryEntry` is not serializable.